supportedBackends/google/pipelines/v2alpha1/src/main/resources/transfer.sh

```diff
@@ -0,0 +1,168 @@
+# The `papi_v2_log` Centaur test is opinionated about the number of log messages around localization/delocalization.
+# The trace logging of `set -x` must be turned off for the `papi_v2_log` test to pass.
+set +x
+
+gsutil_log="gsutil_output.txt"
+
+
+localize_file() {
+  local cloud="$1"
+  local container="$2"
+  local rpflag="$3"
+  # Do not quote rpflag, when that is set it will be -u project which should be two distinct arguments.
+  rm -f "$HOME/.config/gcloud/gce" && gsutil ${rpflag} -m cp "$cloud" "$container" > "$gsutil_log" 2>&1
+}
+
+localize_directory() {
+  local cloud="$1"
+  local container="$2"
+  local rpflag="$3"
+  # Do not quote rpflag, when that is set it will be -u project which should be two distinct arguments.
+  mkdir -p "${container}" && rm -f "$HOME/.config/gcloud/gce" && gsutil ${rpflag} -m rsync -r "${cloud}" "${container}" > "$gsutil_log" 2>&1
+}
+
+# Content type is sometimes (not always) specified for delocalizations.
+gsutil_content_flag() {
+  local content="$1"
+
+  if [[ ! -z "${content}" ]]; then
+    echo -n -- "-h 'Content-Type: $content'"
+  fi
+}
+
+delocalize_file() {
+  local cloud="$1"
+  local container="$2"
+  local rpflag="$3"
+  local content="$4"
+
+  local content_flag=$(gsutil_content_flag $content)
+  # Do not quote rpflag or content_flag, when those are set they will be two distinct arguments each.
+  rm -f "$HOME/.config/gcloud/gce" && gsutil ${rpflag} -m ${content_flag} cp "$container" "$cloud" > "$gsutil_log" 2>&1
+}
+
+delocalize_directory() {
+  local cloud="$1"
+  local container="$2"
+  local rpflag="$3"
+  local content="$4"
+
+  local content_flag=$(gsutil_content_flag $content)
+  # Do not quote rpflag or content_flag, when those are set they will be two distinct arguments each.
+  rm -f "$HOME/.config/gcloud/gce" && gsutil ${rpflag} -m ${content_flag} rsync -r "$container" "$cloud" > "$gsutil_log" 2>&1
+}
+
+timestamped_message() {
+  printf '%s %s\n' "$(date -u '+%Y/%m/%d %H:%M:%S')" "$1"
```

Comment:
Timezone too?

Reply:
This is a copy/paste of the existing code for which we have surprisingly sensitive tests.